src/EnergyPlus/SolarShading.cc

```diff
@@ -372,9 +373,13 @@ void checkShadingSurfaceSchedules(EnergyPlusData &state)
     for (int surfNum = state.dataSurface->ShadingSurfaceFirst; surfNum <= state.dataSurface->ShadingSurfaceLast; ++surfNum) {
         auto &thisSurface = state.dataSurface->Surface(surfNum);
         if (!thisSurface.IsTransparent) continue;
-        if (state.dataGlobal->AnyEnergyManagementSystemInModel &&
-            (EMSManager::isScheduleManaged(state, thisSurface.SchedShadowSurfIndex) || state.dataGlobal->anyPluginsOrCallbacksInModel)) {
-            // Transmittance schedule defintely has an actuator or may have one via python plugin or API
+        // creating some dummy bools here on purpose -- we need to do some renaming and/or consolidate these into a meaningful new global sometime
+        // for now I want the logic to be as readable as possible, so creating shorthand variables makes it very clear
+        bool const anyPlugins = size(state.dataPluginManager->plugins) > 0;
+        bool const runningByAPI = state.dataGlobal->eplusRunningViaAPI;
+        bool const anyEMS = state.dataGlobal->AnyEnergyManagementSystemInModel;
+        if ((anyEMS && EMSManager::isScheduleManaged(state, thisSurface.SchedShadowSurfIndex)) || runningByAPI || anyPlugins) {
```

Comment:
Added the running by API check here and used clearly named const bools to make sure the logic is very obvious.